synapse/federation/federation_server.py

```diff
@@ -212,18 +216,22 @@ async def on_incoming_transaction(
         # accurate as possible.
         request_time = self._clock.time_msec()
 
-        transaction = Transaction(**transaction_data)
-        transaction_id = transaction.transaction_id  # type: ignore
+        transaction = Transaction(
+            transaction_id=transaction_id,
+            destination=destination,
+            origin=origin,
```

Comment:
Previously we pulled `origin` from `transaction_data`, but the comment in `synapse.federation.transport.server` makes me think we should be using the origin we've calculated instead.